eidas_proxy_service/tests/test_models.py

```diff
@@ -0,0 +1,241 @@
+from base64 import b64decode
+from datetime import datetime
+from io import BytesIO
+from pathlib import Path
+from typing import Any, BinaryIO, Dict, Set, cast
+from unittest import mock
+
+from django.core.exceptions import ValidationError
+from django.test import SimpleTestCase
+
+from eidas_proxy_service.models import LevelOfAssurance, LightRequest, LightToken, NameIdFormat, ServiceProviderType
+
+DATA_DIR = Path(__file__).parent / 'data'  # type: Path
+
+
+class AbstractValidityTestCase(SimpleTestCase):
```

Comment:
I tried that but then I lost method autocompletion & parameter checks and Pycharm complained about unresolved attribute references `self.assertRaises`.  I'd like to avoid that.

I now use `t = cast(SimpleTestCase, self); ...; t.assertRaises(...)` in the mixin. Is there a better way?

Reply:
Not sure, I am not a big fan of going through hoops and loops to get editor happy...

Reply:
I've decided to keep `t = cast(SimpleTestCase, self)`. It makes explicit that the mixin is supposed to be used with `SimpleTestCase`.